packages/spektate/src/Validation.ts

```diff
@@ -0,0 +1,349 @@
+ * @param repoName repository name
+ * @param pat personal access token for manifest repo
+ * @param org Azure DevOps organization name
+ * @param project Azure DevOps project name
+ * @param githubUsername GitHub organization or username for the manifest
+ * repository, eg. for github.com/microsoft/bedrock.git, this would be "microsoft"
+ */
+export const verifyManifestRepo = async (
+  repoName: string,
+  pat: string,
+  org: string,
+  project: string,
+  githubUsername?: string
+): Promise<IValidationError | undefined> => {
+  return new Promise<IValidationError | undefined>(async (resolve, reject) => {
+    let repo: IAzureDevOpsRepo | IGitHub | undefined;
+    if (githubUsername && githubUsername !== "") {
+      repo = {
+        reponame: repoName,
+        username: githubUsername
+      };
+      await getGitHubClusterSync(repo, pat)
+        .then(tags => {
+          resolve();
+        })
+        .catch(e => {
+          resolve({
+            message:
+              "Failed to verify manifest repo for cluster sync status. " +
+              e.toString()
+          });
+        });
+    } else if (repoName !== "") {
+      repo = {
+        org,
+        project,
+        repo: repoName
+      };
+      await getADOClusterSync(repo, pat)
+        .then(tags => {
+          resolve();
+        })
+        .catch(e => {
+          resolve({
+            message:
+              "Failed to verify manifest repo for cluster sync status. " +
+              e.toString()
+          });
+        });
+    } else {
+      resolve({
+        message: "Manifest repository could not be recognized. "
+      });
+    }
+  });
+};
+
+/**
+ * Creates an instance of Azure DevOps pipeline
+ * @param org Azure DevOps organization name
+ * @param project Azure DevOps project name
+ * @param token Azure DevOps personal access token
+ */
+export const createPipeline = (
+  org: string,
+  project: string,
+  token?: string
+) => {
+  return new AzureDevOpsPipeline(org, project, token);
+};
+
+/**
+ * Verifies that source repo information is accessible using the source repo access token.
+ * This is used for fetching author data
+ * @param storageAccountName storage account name
+ * @param storageAccountKey storage access key
+ * @param storageTableName storage table name
+ * @param storagePartitionKey storage partition key name
+ * @param sourceRepoAccessToken source repo access token
+ * @param srcPipeline source pipeline
+ * @param hldPipeline hld pipeline
+ * @param manifestPipeline manifest pipeline
+ */
+export const verifySourceRepoAccess = async (
+  storageAccountName: string,
+  storageAccountKey: string,
+  storageTableName: string,
+  storagePartitionKey: string,
+  sourceRepoAccessToken: string,
+  srcPipeline: AzureDevOpsPipeline,
+  hldPipeline: AzureDevOpsPipeline,
+  manifestPipeline: AzureDevOpsPipeline
+): Promise<IValidationError | undefined> => {
+  return new Promise<IValidationError | undefined>(async (resolve, reject) => {
```

Comment:
I don't think you need this top level Promise wrapper. Just use `return` for all the `IValidationError` instead of resolve

Reply:
I originally had it without a promise wrapper, but found that the function calling it wasn't correctly awaiting it if i had just returns. Will try again